scripts/abins/spowdersemiempiricalcalculator.py

```diff
@@ -269,205 +239,379 @@ def _report_progress(msg: str, reporter: Union[None, Progress] = None, notice: b
-        # calculate discrete S for the given quantum order event
-        value_dft = self._calculate_order[order](q2=q2,
-                                                 frequencies=local_freq,
-                                                 indices=local_coeff,
-                                                 a_tensor=self._a_tensors[atom],
-                                                 a_trace=self._a_traces[atom],
-                                                 b_tensor=self._b_tensors[atom],
-                                                 b_trace=self._b_traces[atom])
-
-        # convolve with instrumental resolution
-        broadening_scheme = abins.parameters.sampling['broadening_scheme']
-        _, rebinned_broad_spectrum = self._instrument.convolve_with_resolution_function(frequencies=local_freq,
-                                                                                        bins=self._bins,
-                                                                                        s_dft=value_dft,
-                                                                                        scheme=broadening_scheme)
+        q2 = self._instrument.calculate_q_powder(input_data=self._bin_centres, angle=angle)
 
-        # calculate transition energies for construction of higher order quantum event
-        local_freq, local_coeff = self._calculate_s_over_threshold(s=value_dft, freq=local_freq, coeff=local_coeff)
+        average_a_traces = np.sum([self._powder_data.get_a_traces(k_index) * kpoint.weight
+                                  for k_index, kpoint in enumerate(self._abins_data.get_kpoints_data())],
+                                  axis=0)
+        return self._isotropic_dw(frequencies=self._bin_centres, q2=q2, a_trace=average_a_traces[:, np.newaxis],
+                                  temperature=self._temperature)
 
-        if return_freq:
-            return local_freq, local_coeff, rebinned_broad_spectrum
+    @staticmethod
+    def _isotropic_dw(*, frequencies, q2, a_trace, temperature):
+        """Compute Debye-Waller factor in isotropic approximation"""
+        if temperature < np.finfo(type(temperature)).eps:
+            coth = 1.
         else:
-            return rebinned_broad_spectrum
+            coth = 1.0 / np.tanh(frequencies * CM1_2_HARTREE / (2.0 * temperature * K_2_HARTREE))
+        return np.exp(-q2 * a_trace / 3.0 * coth * coth)
 
-    # noinspection PyUnusedLocal
-    def _calculate_order_one(self, q2=None, frequencies=None, indices=None, a_tensor=None, a_trace=None,
-                             b_tensor=None, b_trace=None):
+    def _get_empty_sdata(self, use_fine_bins: bool = False,
+                         max_order: Optional[int] = None) -> SData:
+        """
+        Initialise an appropriate SData object for this calculation
+        """
+        if use_fine_bins:
+            bin_centres = self._fine_bin_centres
+        else:
+            bin_centres = self._bin_centres
```

Comment:
This might be more concise
```suggestion
        bin_centres = self._fine_bin_centres if use_fine_bins else self._bin_centres
```